### PR TITLE
Move auth config spec to auth backend modules

### PR DESCRIPTION
**Pull request by chrzaszcz**
Just like for modules and services, this keeps it more organized and simplifies future changes of auth handling, e.g. introducing defaults.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_23](https://circleci.com/gh/esl/MongooseIM/93126) / small_tests / d0bbaecd05febc8582d5ac9c8e4c492ed25df66d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93126/small_tests.23.1) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93126/small_tests.23.1/small/ct_run.mongooseim@localhost.2021-12-08_10.43.04/lib.mongooseim.logs/run.2021-12-08_10.43.04/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93160) / small_tests / 011412ef5622c5dfb5000c49281afdb38ec5c48c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93160/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93160/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_10.45.08/lib.mongooseim.logs/run.2021-12-08_10.45.09/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93182) / small_tests / a5234d2eab9c85431dd3f856c36974fa314dd8ce
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93182/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93182/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_10.59.23/lib.mongooseim.logs/run.2021-12-08_10.59.23/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93218) / small_tests / 90f1ea5fa72ea6d72376b025e8525958307614c0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93218/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93218/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_11.05.10/lib.mongooseim.logs/run.2021-12-08_11.05.10/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93234) / small_tests / f708bc730bee45a74d99d5680ab37ac78c80212f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93234/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93234/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_11.42.57/lib.mongooseim.logs/run.2021-12-08_11.42.58/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93265) / small_tests / 69640b1278bc81a3741570ce40b50fa865f04fd6
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93265/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93265/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_11.45.37/lib.mongooseim.logs/run.2021-12-08_11.45.38/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93287) / small_tests / dc96af3607cae5424e442ec5116b7515a004adaa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3437/93287/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3437/93287/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-08_11.50.17/lib.mongooseim.logs/run.2021-12-08_11.50.17/suite.log.html)

File: src/config/mongoose_config_spec.erl [hunk 1]
```diff
@@ -472,31 +467,29 @@ http_handler_required(_) -> [].
 
 %% path: (host_config[].)auth
 auth() ->
+    Items = maps:from_list([{a2b(Method), ejabberd_auth:config_spec(Method)} ||
+                               Method <- all_auth_methods()]),
```
Comment (NelsonVides) [inline src/config/mongoose_config_spec.erl:471]: This is the only part that feels like a hack, and still, it's so much cleaner than before. Have modules do their own configs, yay! 🎉

**Review by NelsonVides — APPROVED** [review]
I'm all in for this, excellent 👌🏽

File: src/config/mongoose_config_spec.erl (hunk 1 above)
Comment (chrzaszcz) [reply src/config/mongoose_config_spec.erl:471]: Yes, I know it's not that clean, but it's the same as for modules and services.

**Pull request merged**